### PR TITLE
Spruce up game and its tests

### PR DESCRIPTION
**Pull request by langner**
Continued work for #347

### CODE REVIEW

**Comment by marcharper** [comment]
Here's the failing test. Otherwise LGTM:

```
======================================================================
FAIL: test_payoff_with_passed_axes (axelrod.tests.unit.test_plot.TestPlot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/tests/unit/test_plot.py", line 208, in test_payoff_with_passed_axes
    msg="Color bar is not in correct location.",
AssertionError: 0.9179031197390574 != 0.8685161708228115 : Color bar is not in correct location.
```

File: axelrod/game.py [hunk 1]
```diff
@@ -1,33 +1,61 @@
 from typing import Tuple, Union
 
-from .action import Action
+from axelrod import Action
```
[the remark this answers is not in the document]
Comment (langner) [reply axelrod/game.py]: Well, relative imports are less common (also in the Axelrod library AFAIR) and generally discouraged. The rationale is mostly that relative imports can be ambiguous, whereas absolute ones are not. For more context: https://www.python.org/dev/peps/pep-0328

**Comment by langner** [comment]
> Here's the failing test. Otherwise LGTM:
> 
> ```
> ======================================================================
> FAIL: test_payoff_with_passed_axes (axelrod.tests.unit.test_plot.TestPlot)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/travis/build/Axelrod-Python/Axelrod/axelrod/tests/unit/test_plot.py", line 208, in test_payoff_with_passed_axes
>     msg="Color bar is not in correct location.",
> AssertionError: 0.9179031197390574 != 0.8685161708228115 : Color bar is not in correct location.
> ```

Is this related at all?

**Comment by marcharper** [comment]
It doesn't appear to be related, see #1207.

@drvinceknight looks like maybe a dependency change broke a test?

**Comment by langner** [comment]
> It doesn't appear to be related, see #1207.
> 
> @drvinceknight looks like maybe a dependency change broke a test?

I'd bet on matplotlib.

**Comment by langner** [comment]
... and seems that #1208 does fix it, so something changed with matplotlib 3.0.0 that broke TestPlot.

File: axelrod/game.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (drvinceknight) [reply axelrod/game.py]: Awesome: thanks for the explanation.

File: axelrod/tests/unit/test_game.py [hunk 2]
```diff
@@ -1,82 +1,81 @@
...
 
+    def test_default_equality(self):
         self.assertEqual(Game(), Game())
 
+    def test_not_default_equality(self):
+        self.assertEqual(Game(1, 2, 3, 4), Game(1, 2, 3, 4))
+        self.assertNotEqual(Game(1, 2, 3, 4), Game(1, 2, 3, 5))
+        self.assertNotEqual(Game(1, 2, 3, 4), Game())
+
+    def test_wront_class_equality(self):
```
Comment (meatballs) [inline axelrod/tests/unit/test_game.py]: Looks like a typo here

**Comment by drvinceknight** [comment]
> I'd bet on matplotlib.

Good call! Have rerun the tests and they all look fine. I've opened https://github.com/Axelrod-Python/Axelrod/issues/1209

File: axelrod/tests/unit/test_game.py (hunk 2 above)
Comment (langner) [reply axelrod/tests/unit/test_game.py]: Fixed and rebased.

**Pull request merged**